core/tracing/hooks.go

```diff
@@ -214,6 +214,44 @@ const (
 	BalanceDecreaseSelfdestructBurn BalanceChangeReason = 14
 )
 
+// String returns a string representation of the reason.
+func (r BalanceChangeReason) String() string {
```

Comment:
Please use <https://pkg.go.dev/golang.org/x/tools/cmd/stringer> to generate this method.